### PR TITLE
Recipe improvements

### PR DESCRIPTION
**Pull request by pplantinga**
Prepare challenge and other recipe improvements.

I think the main one to note is the change from chunking done in the `train.py` to chunking done in `prepare_neuro.py` which I think will lead to more consistent results, and cleans up the file considerably.

### CODE REVIEW

**Comment by bcordel** [comment]
Hey Peter changes look great thanks a bunch! Wondering about the overlapped chunked evaluation, do you want to move that to a different file? 

Also the PRs are a great idea, I think this'll be a lot easier to keep track of. 

Oh and it is 6:30pm on a Friday so absolutely no rush to answer, I just read through this now and thought I'd write while its fresh!

**Comment by pplantinga** [comment]
> Hey Peter changes look great thanks a bunch! Wondering about the overlapped chunked evaluation, do you want to move that to a different file?

This is a good question. I think ideally we would track all the predictions made on the validation and test set chunks and save to a file for further analysis. Then we can experiment with combining chunks in different ways on the validation set and check final performance on the test set. 

Some function could also be added to the end of the training script or even during validation to do a basic chunk combination and print performance for different categories, just for eyeballing the results while tweaking the recipe. Personally I'm interested in the difference between a Whisper feature encoder and WavLM feature encoder for different languages and tasks, so I'd like the recipe at the end to break down performance (f-score, precision, recall) by language and task.

**Comment by pplantinga** [comment]
Let me take a crack at adding these
